datadog/data_source_datadog_rum_application.go

```diff
@@ -0,0 +1,109 @@
+package datadog
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDatadogRUMApplication() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this data source to retrieve a Datadog RUM Application.",
+		ReadContext: dataSourceDatadogRUMApplicationRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name used to search for a RUM application",
+			},
+			"type_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The type used to search for a RUM application",
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "ID of the RUM application. If set, this takes precedence over name and type filters.",
```

Comment:
We should add ConflictsWith (at least here?) to make it more obvious it doesn't work with filters.